Windows build error with ONNX Runtime: Ort::Session expects const wchar_t* for model path

### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.


### Ultralytics YOLO Component

_No response_

### Bug

## Title

Windows build error with ONNX Runtime: `Ort::Session` expects `const wchar_t*` for model path

### Description

Hi, thank you for the great C++ examples!

I encountered a build issue when compiling the `examples/cpp/ONNXRuntime` example on Windows.



### Error

Compilation fails at the following line:

```cpp
session_ = new Ort::Session(env_, config_.model_path.c_str(), session_options_);
```

The compiler reports:

```
error C2665: 'Ort::Session::Session':
no overloaded function could convert all the argument types

cannot convert parameter 2 from
'const char *' to 'const wchar_t *'
```

On Windows, the ONNX Runtime C++ API expects a `const wchar_t*` model path, while `config_.model_path` is a `std::string`.

### Suggested solution

Convert the model path to `std::wstring` on Windows before creating the session.

For example:

```cpp
#ifdef _WIN32
std::wstring model_path(config_.model_path.begin(), config_.model_path.end());
session_ = new Ort::Session(env_, model_path.c_str(), session_options_);
#else
session_ = new Ort::Session(env_, config_.model_path.c_str(), session_options_);
#endif
```

This should keep compatibility with Linux while allowing the example to compile successfully on Windows.

Please let me know if this is the expected behavior or if there is a better solution.

Thanks!


### Environment

### Environment

* OS: Windows 11
* Visual Studio: Visual Studio 2026
* CMake: 3.x
* ONNX Runtime: 1.16.1
* OpenCV: 4.x

### Minimal Reproducible Example

```cpp
#ifdef _WIN32
std::wstring model_path(config_.model_path.begin(), config_.model_path.end());
session_ = new Ort::Session(env_, model_path.c_str(), session_options_);
#else
session_ = new Ort::Session(env_, config_.model_path.c_str(), session_options_);
#endif
```

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

--- a/examples/cpp/ONNXRuntime/inference.cpp
+++ b/examples/cpp/ONNXRuntime/inference.cpp
@@ -3,6 +3,7 @@
 #include "inference.h"
 
 #include <array>
+#include <filesystem>
 #include <iostream>
 #include <regex>
 
@@ -77,7 +78,8 @@ Predictor::Predictor(const Config& config) : config_(config) {
     }
 #endif
 
-    session_ = new Ort::Session(env_, config_.model_path.c_str(), session_options_);
+    const std::filesystem::path model_path = std::filesystem::u8path(config_.model_path);
+    session_ = new Ort::Session(env_, model_path.c_str(), session_options_);
 
     Ort::AllocatorWithDefaultOptions allocator;
     for (size_t i = 0; i < session_->GetInputCount(); ++i) {

